### PR TITLE
Closes #2230: add AddDataset permissions into Solr

### LINKED ISSUE #2230: Add information about AddDataset permission for dataverses in Solr index

### PR DESCRIPTION
**Pull request by wtr-icm**
MR is divided in two commits:
First is for cleaning of existing code
Second for proper task changes

### CODE REVIEW

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/RoleAssignmentRepository.java [hunk 1]
```diff
@@ -55,4 +61,17 @@ public int deleteAllByAssigneeIdentifier(String identifier) {
...
+        }
+        String query = "SELECT id FROM dvobject WHERE dtype = 'Dataverse' " +
+                "and id in (select definitionpoint_id from roleassignment " +
+                "where assigneeidentifier in ("
+                + identifiers.stream().map(i -> "'" + i + "'").collect(Collectors.joining(",")) + "));";
+        logger.info("query: {}", query);
+        Query nativeQuery = em.createNativeQuery(query);
+        return (List<Integer>) nativeQuery.getResultList();
+    }
 }
```
Comment (tomaszblazejczyk) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/RoleAssignmentRepository.java:77]: Why didn't you use NamedQuery as above?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/PermissionsSolrDoc.java [hunk 2]
```diff
@@ -55,27 +59,36 @@ public String getSolrId() {
...
     }
 
     // -------------------- toString --------------------
 
     @Override
     public String toString() {
         return "PermissionsSolrDoc [dvObjectId=" + dvObjectId + ", solrId=" + solrId + ", datasetVersionId="
-                + datasetVersionId + ", nameOrTitle=" + nameOrTitle + ", permissions=" + permissions + "]";
+                + datasetVersionId + ", nameOrTitle=" + nameOrTitle + ", searchPermissions=" + searchPermissions
+                + ", addDatasetPermissions=" + "]";
```
Comment (tomaszblazejczyk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/PermissionsSolrDoc.java]: Missed "addDatasetPermission" content?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDataverseCommandTest.java [hunk 3]
```diff
@@ -226,7 +226,7 @@ public void testDefaultOptions() {
         assertEquals(dv, roleAssignment.getDefinitionPoint());
         assertEquals(roleAssignment.getAssigneeIdentifier(), request.getUser().getIdentifier());
 
-        // The following is a pretty wierd way to test that the create date defaults to 
+        // The following is a pretty wierd way to test that the create date defaults to
```
Comment (tomaszblazejczyk) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDataverseCommandTest.java]: Missed typo in this line (wierd -> weird)?

**Review by tomaszblazejczyk — APPROVED** [review]

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/RoleAssignmentRepository.java (hunk 1 above)
Comment (wtr-icm) [reply dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/user/RoleAssignmentRepository.java:77]: Unfortunately this method uses native query with IN clause, which prevents us from use `setParameter` (as this does not work with lists for native queries) and that part of query has to be interpolated, so we would not have much benefit from NamedQuery.
As this method is only moved from other place I didn't wanted to spend too much time here, but in fact it should be rewritten in order to remove that horrible concatenation of ids. On the other hand it is used only in one place, and probably the whole part is rarely (if ever) used.

**Pull request merged**